File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/writing_data/sample5.cpp
#include <dv-processing/data/generate.hpp>
#include <dv-processing/io/mono_camera_writer.hpp>

int main() {
	// IMU data only configuration
	auto config = dv::io::MonoCameraWriter::Config("DVXplorer_sample");
	config.addImuStream();

	// Create the writer instance, it will only have a single IMU data output stream
	dv::io::MonoCameraWriter writer("mono_writer_sample.aedat4", config);

	// Write 100 IMU measurements
	for (int i = 0; i < 100; i++) {
		// Generate some monotonically increasing timestamp
		const int64_t timestamp = i * 1000;

		// Some sample measurements - no rotation on gyro and a one-G gravity on Y axis of accelerometer
		const dv::IMU measurement = dv::data::generate::levelImuWithNoise(timestamp);

		// Write the measurement
		writer.writeImu(measurement);
	}

	return 0;
}

File: source_code/ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/writing_data/sample5.py
import dv_processing as dv

# IMU data only configuration
config = dv.io.MonoCameraWriter.Config("DVXplorer_sample")
config.addImuStream()

# Create the writer instance, it will only have a single IMU data output stream
writer = dv.io.MonoCameraWriter("mono_writer_sample.aedat4", config)

# Write 100 IMU measurements
for i in range(100):
    # Generate some monotonically increasing timestamp
    timestamp = i * 1000

    # Single IMU measurement instance
    measurement = dv.data.generate.levelImuWithNoise(timestamp)

    # Write the measurement
    writer.writeImu(measurement)
